eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -210,6 +210,16 @@ jobs:
     displayName: Publish Code Coverage to DevOps
     condition: eq(variables['AZ_SDK_CODE_COV'], 1)
 
+  - task: mspremier.BuildQualityChecks.QualityChecks-task.BuildQualityChecks@6
+    displayName: Check coverage
+    inputs:
+      checkCoverage: true
+      coverageFailOption: fixed
+      coverageType: line
+      # 85% minimum line coverage
+      coverageThreshold: 85
```

Comment:
Can we add a check for branch coverage as well (and not just line coverage), which arguably is more important (or at least is a bit low atm).

Reply:
branches would be hard to beat. 
XD
consider an example of one method that takes an enum as arg.
Then inside the method we use switch for the enum. 
Compiler requires us to use a default for the switch, but if the switch already cover all enum values then how do you test the default behavior without changing the code...
That´s just one example, there are hard cases around our code of things that he code itself is blocking for happening but it is still another branch in there

Reply:
I understand there are always going to be some edge cases like the ones you mentioned. That just means we set a reasonable bar (somewhere between 80 and 95%). I don't think that means we shouldn't add it to the bar all together. We should start with the coverage we have at the moment and raise it up to that reasonable bar as we improve coverage.

Reply:
Our current branch coverage is ~64% according to the coverage report: https://dev.azure.com/azure-sdk/public/_build/results?buildId=339456&view=codecoverage-tab

That number is already being published. We should only have a step to enforce that if we want to fail the pipeline for branch coverage falling below a threshold.

Reply:
> We should only have a step to enforce that if we want to fail the pipeline for branch coverage falling below a threshold.

I believe we should, just like line coverage. We can start off by setting it to 60 (with the goal to move it up by adding missing tests).